cmake/plugin.cmake

```diff
@@ -78,9 +78,13 @@ MACRO(MYSQL_ADD_PLUGIN)
     IF (ARG_DISABLED)
       SET(howtobuild NO)
     ELSEIF (compat STREQUAL ".")
-      SET(howtobuild DYNAMIC)
+      IF(ARG_STATIC_ONLY)
+        SET(howtobuild STATIC)
+      ELSE()
+        SET(howtobuild DYNAMIC)
+      ENDIF()
     ELSEIF (compat STREQUAL "with.")
-      IF (NOT ARG_MODULE_ONLY)
+      IF (NOT ARG_MODULE_ONLY OR ARG_STATIC_ONLY)
```

Comment:
are you handling the case where both MODULE_ONLY and STATIC_ONLY are given, for the same plugin?

Reply:
MODULE_ONLY excludes information_schema (except wsrep_info) and replication plugins, built in storage engines (partition, archive, blackhole, heap,myisammrg, perfschema, sphinx, sequence). This implies a compile link to the server in the plugin.cmake and enabling instrumentation.I'm struggling to see a consistent purpose. Can you enlighten me?

I made the assumption that ARG_STATIC_ONLY is the stronger indicator that it should be static only. I haven't found any with MODULE_ONLY and STATIC_ONLY. Do you know how they should be handled?

Reply:
MODULE_ONLY - build only as module. usually, but not always, a module it is the same as shared library, that is used via dlopen/dlsym , or LoadLibrary/GetProcAddress .  Some systems make a distinction between shared libraries, and modules. On Mac for example, module is .so, vs shared library which is .dylib

STATIC_ONLY - only build it as static library, linked into the server

MODULE_ONLY and STATIC_ONLY as obviously mutually exclusive (though perhaps there is no check for that in the MYSQL_ADD_PLUGIN)

Reply:
module is the same as "dynamic" in MySQL plugin parlance

Reply:
Thank you. I was looking at writing something more extensive on the number of checks of user specified incompatible with plugin definition, mutually exclusive options (now I've got another), but wasn't going to try to push that to 10.1 (and maybe this shouldn't either).